04_documentation/code_documentation_tools_exercise.md

```diff
@@ -0,0 +1,92 @@
+            - Add the `how-to.md` with label `Getting Started`
+          You should only have two buttons "Home" and "User Guide" in the generated website.
+    - Add the [repository URL](https://www.mkdocs.org/user-guide/configuration/#repo_url). This adds a "Edit on GitHub" button at the top right of the website. Try if this works. If it does not work, this is because older versions of MkDocs set a link to the `master` branch by default, but GitHub calls this branch `main` nowadays. In order to point to the correct branch, set the `edit_uri` setting to `edit/main/docs/`.
+
+        **Note** The "Edit on GitHub" also fails if you have not pushed the corresponding Markdown file to your repository yet.
+
+    - Make yourself the [website's author](https://www.mkdocs.org/user-guide/configuration/#site_author). You can use your real name or your GitHub username. This setting will not change the appearance of your website, but adds the name to the HTML metadata.
+- [Deploy your website](https://www.mkdocs.org/user-guide/deploying-your-docs/) to [GitHub Pages](https://pages.github.com/). This consists of following steps:
+    - Run `mkdocs gh-deploy` to build and deploy the website:
+
+      Deployment of the website will create a branch `gh-pages` inside the repository if it does not already exist. This branch contains all files needed to deliver the website. You should not edit any of the files in the branch manually since `mkdocs gh-deploy` will overwrite all changes. Instead, fix problems in your main branch and deploy the website afterwards again.
+
+    - Verify that the "Pages" feature of your repository has been activated. When you are on your repository on GitHub go to "Settings" (at top of window) -> "Pages". It should show an information box " Your site is ready to be published at URL" with the URL to the website of your documentation.
+
+      If the "Pages" feature has not been activated automatically, choose the `gh-pages` branch as source and save/activate the website.
+
+      Check the website. If you get a "404" error
+
+      ```text
+      404
+
+      There isn't a GitHub Pages site here.
+      ```
+
+      wait for 5-10 minutes and try again. It takes a while for the website to become visible.
+
+    - Add the URL of the documentation to your GitHub project. Go to your repository on GitHub and edit the "About" section at the top-right corner. For editing clock on the small gear symbol. Add the URL in the box "Website".
+- Verify that your website looks as expected and that the "Edit on GitHub" feature works for all pages.
+- Submit your documentation website via an issue. You cannot submit a pull request since you cannot use a fork in the exercise. You can also add the issue before you are finished in order to ask question. In that case please mention us in the issue such that we get notified. Please write a short note in the issue when you are done and the issue is ready to review. **TODO** Skip this for testing the exercise.
```

Comment:
```suggestion
- Submit your documentation website via an issue. You cannot submit a pull request since you cannot use a fork in the exercise. You can also add the issue before you are done in order to ask questions. In that case please mention us in the issue such that we get notified. Please write a short note in the issue when you are done and the issue is ready to review. **TODO** Skip this for testing the exercise.
```